lsp/src/saros/lsp/editor/Editor.java

```diff
@@ -0,0 +1,29 @@
+package saros.lsp.editor;
+
+import java.io.IOException;
+import java.io.InputStream;
+import org.apache.commons.io.IOUtils;
+import org.eclipse.lsp4j.TextDocumentItem;
+import saros.filesystem.IFile;
+
+public class Editor extends TextDocumentItem {
+
+  public Editor(TextDocumentItem documentItem) {
+    this(documentItem.getText(), documentItem.getUri());
+    super.setVersion(documentItem.getVersion());
+    super.setLanguageId(documentItem.getLanguageId());
+  }
+
+  public Editor(String text, String uri) {
+    super.setUri(uri);
+    super.setText(text);
+  }
+
+  public Editor(IFile file) throws IOException {
+    super.setUri("file:///" + file.getReferencePointRelativePath().toString());
```

Comment:
I do no get this. You are building an absolute URL out of some relative location. Where do the magic happen aka where do you get the absolute path to the some root directory ?